io-sim-classes/src/Control/Monad/Class/MonadThrow.hs

```diff
@@ -118,31 +139,72 @@ class MonadCatch m => MonadMask m where
 
 instance MonadThrow IO where
 
-  throwM   = IO.throwIO
+  throwM   = throwIO
 
-  bracket  = IO.bracket
-  bracket_ = IO.bracket_
-  finally  = IO.finally
+  bracket  = E.bracket
```

Comment:
Wait what? No, the whole point here is the IO instance is exactly the `base` functions.

And I don't see any reason we need anything here? It's only `generalBracket` that needs to be defined here for IO.